src/conditionals/programs_with_decision_points/hard/q14/MainTest.java

```diff
@@ -0,0 +1,73 @@
+package conditionals.programs_with_decision_points.hard.q14;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    
+    @Override
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "apple\nbanana";
+        
+        return new Clause[]{
+                new StringLiteral("Enter word1 "),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("Enter word2 "),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+    
+    /**
+     * Provides test cases as (word1, word2, expectedPhrase).<br>
+     * The expected outputs are defined according to:
+     * <ul>
+     *    <li>Combined length < 10: "Short phrase."</li>
+     *    <li>Combined length between 10 and 15 (inclusive): "Medium phrase."</li>
+     *    <li>Combined length > 15: if word1 starts with lowercase 'a', "a-start long phrase." otherwise "Long phrase."</li>
+     * </ul>
+     */
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
```

Comment:
same as last pr, you should also add tests on either side of the boundaries for each threshold